pallets/gear/src/tests.rs

```diff
@@ -4139,6 +4373,29 @@ mod utils {
             .expect("can't find message send event")
     }
 
+    pub(super) fn maybe_last_message(account: AccountId) -> Option<StoredMessage> {
+        SystemPallet::<Test>::events()
+            .iter()
+            .rev()
+            .filter_map(|r| {
+                if let MockEvent::Gear(e) = r.event.clone() {
+                    Some(e)
```

Comment:
Better to clone inside if it is possible.
```suggestion
                if let MockEvent::Gear(e) = r.event {
                    Some(e.clone())
```

Reply:
would be like

```rust
  pub(super) fn maybe_last_message(account: AccountId) -> Option<StoredMessage> {                                                                                                                                                                                             
        SystemPallet::<Test>::events()                                                                                                                                                                                                                                          
            .into_iter()                                                                                                                                                                                                                                                        
            .rev()                                                                                                                                                                                                                                                              
            .find_map(|e| match e.event {                                                                                                                                                                                                                                       
                MockEvent::Gear(Event::UserMessageSent { message, .. }) => {                                                                                                                                                                                                    
                    if message.destination() == account.into() {                                                                                                                                                                                                                
                        Some(message)                                                                                                                                                                                                                                           
                    } else {                                                                                                                                                                                                                                                    
                        None                                                                                                                                                                                                                                                    
                    }                                                                                                                                                                                                                                                           
                }                                                                                                                                                                                                                                                               
                _ => None,                                                                                                                                                                                                                                                      
            })                                                                                                                                                                                                                                                                  
    }     
```